include/zephyr/rtio/rtio.h

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Real-Time IO device API for moving bytes with low effort
+ *
+ * RTIO uses a SPSC lock-free queue pair to enable a DMA and ISR friendly I/O API.
```

Comment:
I had rtio_fifo as the name before, but thought I might want to add a multi-producer single consumer queue for the completion queue given it can potentially accept completions from multiple producers (chains).

I'm open to suggestions on the best route for naming these things. SPSC, MPSC, MPMC are pretty common terms for lock free queues that I've seen though.

Reply:
I am fine with SPSC as long as it is defined. I spent a lot of years in storage and then some time in networking. The acronyms that I found common in each of these were common within that particular industry but not so much outside. People should feel included when reading this and not like they have to solve a puzzle.  If I Google "DMA computer" I will get Direct Memory Access as the first entry. "ISR computer" gives me the right answer in the first entry.  "SPSC computer" not so much... I have gone pages through the results and get no relevent hits that says that means single-producer single-consumer.

Reply:
That is entirely fair and unfortunate its a buried term